docs/DeveloperGuide.md

```diff
@@ -1,38 +1,591 @@
+saving it to the file.
+
+3. The user closes the application and relaunches it. The PINHandler loads the saved PIN and authentication 
+enabled status from the file again.
+
+4. The user attempts to log in by entering their PIN. The authenticate method hashes the entered PIN and 
+compares it with the saved hashed PIN. If they match, the user is successfully authenticated.
+
+5. The user decides to reset their PIN by entering their current PIN and creating a new one using the resetPin 
+method.
+
+6. The user disables authentication upon startup using the 'pin disable' command. The authenticationEnabled flag 
+is set to false and saved to the file.
+
+7. The user relaunches the application, and authentication is no longer required since it has been disabled. 
+The user can proceed with the application and do any actions without entering a PIN.
+
+Code Snippet
+```
+// Initialize PINHandler
+PINHandler pinHandler = new PINHandler();
+
+// Check if authentication is enabled
+if (PINHandler.getAuthenticationStatus()) {
+// Authenticate user
+PINHandler.authenticate();
+} else {
+// Authentication is disabled, proceed with application logic
+}
+```
+
+<ins> Design Considerations </ins>
+
+Resetting PIN: The resetPin() method allows users to change their PIN by first verifying their current PIN. This adds
+an extra layer of security to prevent unauthorized PIN changes.
+
+Authentication Management: Users have the option to enable or disable authentication upon startup using the 'pin enable'
+and 'pin disable' commands. This flexibility allows users to customize their authentication preferences based on their
+security needs and convenience.
+
+### Chart
+
+<ins>Overview</ins>
+
+The Chart class is responsible for generating a visual representation of the transaction solution in the form of a chart.
+
+The chart is displayed in a separate window and shows the individual balances among the group members.
+
+This is handled using a package called 'XChart'.
+
+It provides a convenient way to visualize data, particularly for member balances in the LongAh application. The bar chart generated by this class showcases positive and negative balances of members, offering insights into financial statuses.
+
+<ins>Implementation Details</ins>
+
+Data Representation: 
+
+The Chart class utilizes the XChart library to represent data in the form of bar charts. It distinguishes positive and negative balances by differentiating them with green and red colors, respectively.
+
+
+<ins>Class Structure</ins>
+
+The Chart class consists of the following components:
+
+- *Constructor*: Instantiates a new Chart object by displaying the provided category chart.
+
+- *Static Method*: viewBalancesBarChart constructs and displays a bar chart representing member balances.
+
+<ins>Methods</ins>
+
+- *viewBalancesBarChart*(List<String> members, List<Double> balances): Generates a bar chart displaying member balances. It 
```

Comment:
Most methods shown do not include the variables (if have). Perhaps can standardize if including the variables or not.